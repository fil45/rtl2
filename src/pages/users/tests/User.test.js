import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import User from "../User";

const deleteFunction = jest.fn();

const user = {
  id: 1,
  firstname: "Alex",
  lastname: "Hodovas",
};

const component = <User user={user} onDelete={deleteFunction} />;

describe("User page --->", () => {
  it("should render the page", () => {
    const { getByText } = render(component);
    const firstname = getByText(/Alex/i);
    expect(firstname).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(component);
    expect(asFragment(component)).toMatchSnapshot();
  });

  it("should delete button works", async () => {
    const { getByRole } = render(component);

    const deleteButton = getByRole("button");
    expect(deleteButton).toBeInTheDocument();
    userEvent.click(deleteButton);

    expect(deleteFunction).toBeCalledWith(1);
    expect(deleteFunction).toBeCalledTimes(1);
  });
});
