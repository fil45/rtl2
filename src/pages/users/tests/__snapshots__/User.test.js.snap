// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User page ---> should match snapshot 1`] = `
<DocumentFragment>
  <li>
    <p>
      1 Alex Hodovas
    </p>
    <button>
      Delete
    </button>
  </li>
</DocumentFragment>
`;
